api4/job.go

```diff
@@ -134,12 +140,19 @@ func getJobs(c *Context, w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	if !c.App.SessionHasPermissionTo(*c.App.Session(), model.PERMISSION_READ_JOBS) {
-		c.SetPermissionError(model.PERMISSION_READ_JOBS)
-		return
+	var validJobTypes []string
+	for _, jobType := range model.ALL_JOB_TYPES {
+		hasPermission, permissionRequired := c.App.SessionHasPermissionToReadJob(*c.App.Session(), jobType)
+		if permissionRequired == nil {
+			mlog.Warn("The job types of a job you are trying to retrieve does not contain permissions", mlog.String("jobType", jobType))
+			continue
+		}
+		if hasPermission {
+			validJobTypes = append(validJobTypes, jobType)
+		}
 	}
 
-	jobs, err := c.App.GetJobsPage(c.Params.Page, c.Params.PerPage)
+	jobs, err := c.App.GetJobsByTypesPage(validJobTypes, c.Params.Page, c.Params.PerPage)
```

Comment:
There were two ways I could have implemented this. 
1. Determine JobTypes session has access to, then query for those JobTypes.
- Good - (Performance) Allows the database to handle filtering (there IS an index on type)
- Bad - (Maintenance) Requires maintaining model.ALL_JOB_TYPES.
2. Return all Jobs and Loop
- Good - (Maintenance) Less changes, will work automatically with new JobTypes
- Bad - (Performance) Although done in paging, all Jobs are returned therefore could be looping through thousands of jobs.

Additional Information - The GetJobs() endpoint is not used by our Web App.